12-project-maintenance.Rmd

```diff
@@ -8,3 +8,168 @@ ottrpal::set_knitr_image_path()
 ```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Major point!! example image"}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21a84b32106_0_58")
 ```
+
+## Understand best practices for maintaining projects
+
+Many R projects are either maintained long term or they deprecate. Many folks have many R projects over time which means keeping all your R projects maintained could easily become a time consuming endeavor!
+
+In order to keep maintenance a priority without it using up all your time, we have a few tips to discuss. These tips will help you ease the burden of maintenance and lead to a healthier R project long term.
+
+Building some of the following infrastructure will take a bit more time upfront, but will save you time overall:
+
+1. Have unit testing with `testthat`
+2. Use continuous deployment / continuous integration (CI/CD) principles
+
+## What is Unit Testing?
+
+One of the most comprehensive methods for making sure your code works is building tests. Unit testing can be really straightforward to build using the [`testthat` package](https://testthat.r-lib.org/). `testhat` works with the `usethis` R package so you can start by installing both of those.
+
+```
+install.packages(c("usethis", "testthat"))
+```
+
+In a previous chapter we talked about making [custom functions](https://hutchdatascience.org/Tools_for_Reproducible_Workflows_in_R/functions.html) and ended up showing an example called `make_shades()`.
+
+
+To create tests for this function we can create a new file for it. We can do this by calling `use_test` and ideally we will call it something that relates to what function is being tested there.
+
+```
+usethis::use_test("make_shades")
+```
+
+This will open up a page that looks like this. This is the template test that shows up when you make a new test.
```

Comment:
```suggestion
This will open up a page that looks like this. This is the template test that shows up when you make a new test. It will have an explanation of how one might test `2*2`.
```

Reply:
adding more explicit explanation